jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6132 @@
+	 * @param exchangeValue value that will be set in obj at offset if compare is successful
+	 * @return value of the field before this operation. This will be compareValue if the exchange was successful
+	 */
+	public final Object compareAndExchangeObjectRelease(Object obj, long offset, Object compareValue,
+			Object exchangeValue) {
+		return compareAndExchangeObject(obj, offset, compareValue, exchangeValue);
+	}
+
+	/**
+	 * Sets the parameter value at offset in obj if the compare value 
+	 * matches the existing value in the object.
+	 * The get operation has memory semantics of get.
+	 * The set operation has the memory semantics of set.
+	 *
+	 * @param obj object into which to store the value
+	 * @param offset location to compare and store value in obj
+	 * @param compareValue value that is expected to be in obj at offset
+	 * @param setValue value that will be set in obj at offset if compare is successful
+	 * @return boolean value indicating whether the field was updated
+	 */
+	public final boolean weakCompareAndSetObjectPlain(Object obj, long offset, Object compareValue, Object setValue) {
+		return compareAndSetObject(obj, offset, compareValue, setValue);
+	}
+
+	/**
+	 * Sets the parameter value at offset in obj if the compare value 
+	 * matches the existing value in the object.
+	 * The get operation has memory semantics of getAcquire.
+	 * The set operation has the memory semantics of set.
+	 *
+	 * @param obj object into which to store the value
+	 * @param offset location to compare and store value in obj
+	 * @param compareValue value that is expected to be in obj at offset
+	 * @param setValue value that will be set in obj at offset if compare is successful
+	 * @return boolean value indicating whether the field was updated
+	 */
+	public final boolean weakCompareAndSetObjectAcquire(Object obj, long offset, Object compareValue, Object setValue) {
+		return compareAndSetObject(obj, offset, compareValue, setValue);
+	}
+
+	/**
+	 * Sets the parameter value at offset in obj if the compare value 
+	 * matches the existing value in the object.
+	 * The get operation has memory semantics of get.
+	 * The set operation has the memory semantics of setRelease.
+	 *
+	 * @param obj object into which to store the value
+	 * @param offset location to compare and store value in obj
+	 * @param compareValue value that is expected to be in obj at offset
+	 * @param setValue value that will be set in obj at offset if compare is successful
+	 * @return boolean value indicating whether the field was updated
+	 */
+	public final boolean weakCompareAndSetObjectRelease(Object obj, long offset, Object compareValue, Object setValue) {
+		return compareAndSetObject(obj, offset, compareValue, setValue);
+	}
+
+	/**
+	 * Sets the parameter value at offset in obj if the compare value 
+	 * matches the existing value in the object.
+	 * The get operation has memory semantics of get.
+	 * The set operation has the memory semantics of set.
+	 *
```

Comment:
This is the same as Plain, should it be different?

Reply:
They are the same in `VarHandle`.